task_sdk/tests/api/test_client.py

```diff
@@ -82,6 +85,88 @@ def make_client(transport: httpx.MockTransport) -> Client:
     return Client(base_url="test://server", token="", transport=transport)
 
 
+class TestTaskInstanceOperations:
+    """
+    Test that the TestVariableOperations class works as expected. While the operations are simple, it
+    still catches the basic functionality of the client for task instances including endpoint and
+    response parsing.
+    """
+
+    def test_task_instance_start(self):
+        # Simulate a successful response from the server that starts a task
+        ti_id = uuid6.uuid7()
+
+        def handle_request(request: httpx.Request) -> httpx.Response:
+            if request.url.path == f"/task-instances/{ti_id}/state":
+                return httpx.Response(
+                    status_code=204,
+                )
```

Comment:
We should probably assert something about the request body/payload here I think.

Reply:
Yeah makes sense. Adding some validation

Reply:
Yeah makes sense. Adding some validation